website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+vertex X and Y position floats!
+
+
+To fix this, we just need to go back to our `VertexAttribPointer()`, which is below our shader compiling section, and
+change it to:
+```c#
+//                       3 floats for position + 2 floats for texture coordinates! \/
+_gl.VertexAttribPointer(positionLoc, 3, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)0);
+```
+The value `3` in the seccond parameter are
+
+This will make the buffer be read like this (including the texture coordinate pointer, which we'll add into our code later):
+
+![Buffer reading right data](../../../images/opengl/chapter1/lesson3/right-pointers.png)
+
+Now the positions of each vertex are being read correctly! Each position is still three floats, but by skipping the two floats after each position,
+we skip over the U and V floats in between each XYZ floats, thus reading them correctly.
+
+If we run the program now, we can see our quad again!
+
+## Using texture coordinates
+Now, we just need to prepare the shader to receive the texture coordinate values. For that, we need to modify both our shaders.
+
+Vertex Shader:
+```
+#version 330 core
+
+layout (location = 0) in vec3 aPosition;
+// Add an new input attribute for the texture coordinates
+layout (location = 1) in vec2 aTextureCoord;
+
+// Add an output variable to pass the texture coordinate to the fragment shader
+// This line stores the data that we want to be received by the fragment
+out vec2 frag_texCoords;
+
+void main()
+{
+    gl_Position = vec4(aPosition, 1.0);
+    // Assigin the texture coordinates without any modification to be recived in the fragment
+    frag_texCoords = aTextureCoord;
+}";
+```
+in this case, we are putting manually the `location` of `aTextureCoord` as 1, because we already have a attribute in the location 0.
+
+
+Fragment Shader:
+```
+#version 330 core
+
+// Receive the input from the vertex shader in an attribute
+in vec2 frag_texCoords;
+
+out vec4 out_color;
+
+void main()
+{
+    // This will allow us to see the texture coordinates in action!
+    out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0.0, 1.0);
+}
+```
+As you already know, the Vertex Shader run for every vertex of the mesh and the Fragment Shader run for every pixel inside the mesh.
+This conversion looks weird in first look, but the data comunication between vertex and fragment shader are surely possible because
+OpenGL, automatcally, interpolate the data recived in the Vertex Shader when it's received by the Fragment.
+
+Now we just need to assign the correct layout for `aTextureCoord`. You should add these lines below our first `VertexAttribPointer` call:
+
+```c#
+const uint texCoordLoc = 1;
+_gl.EnableVertexAttribArray(texCoordLoc);
+_gl.VertexAttribPointer(texCoordLoc, 2, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)(3 * sizeof(float)));
+```
+
```

Comment:
```suggestion

Here we're specifying a new vertex attribute, which consists of two floats and repeats every 20 bytes (`5 * sizeof(float)` = 20, that's the size of one vertex).

```

Reply:
This has been added but GitHub didn't pick it up for some reason. Marking as resolved, please shout at me if this was wrong @ThomasMiz!